pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -13,55 +17,77 @@ import (
 )
 
 type Manager struct {
-	client     client.Client
-	nodeName   string
-	policyName types.NamespacedName
-	logger     logr.Logger
+	client   client.Client
+	policy   nmstatev1alpha1.NodeNetworkConfigurationPolicy
+	nodeName string
+	logger   logr.Logger
 }
 
-func NewManager(client client.Client, nodeName string, policyName types.NamespacedName) Manager {
-	return Manager{
-		client:     client,
-		nodeName:   nodeName,
-		policyName: policyName,
-		logger:     logf.Log.WithName("policy/conditions/manager").WithValues("node", nodeName, "policy", policyName),
+func NewManager(client client.Client, nodeName string, policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) Manager {
+	manager := Manager{
+		client:   client,
+		policy:   policy,
+		nodeName: nodeName,
 	}
+	manager.logger = logf.Log.WithName("policy/conditions/manager").WithValues("enactment", nmstatev1alpha1.EnactmentKey(nodeName, policy.Name).Name)
+	return manager
 }
 
 func (m *Manager) NotifyProgressing() {
-	err := m.updateEnactmentCondition(setEnactmentProgressing, "Applying desired state")
+	err := m.updateEnactmentConditions(setEnactmentProgressing, "Applying desired state")
 	if err != nil {
 		m.logger.Error(err, "Error notifying state Progressing")
 	}
 }
 func (m *Manager) NotifyFailedToConfigure(failedErr error) {
-	err := m.updateEnactmentCondition(setEnactmentFailedToConfigure, failedErr.Error())
+	err := m.updateEnactmentConditions(setEnactmentFailedToConfigure, failedErr.Error())
 	if err != nil {
 		m.logger.Error(err, "Error notifying state FailingToConfigure")
 	}
 }
 
 func (m *Manager) NotifySuccess() {
-	err := m.updateEnactmentCondition(setEnactmentSuccess, "successfully reconciled")
+	err := m.updateEnactmentConditions(setEnactmentSuccess, "successfully reconciled")
 	if err != nil {
 		m.logger.Error(err, "Error notifying state Success")
 	}
 }
 
-func (m *Manager) updateEnactmentCondition(
-	conditionsSetter func(*nmstatev1alpha1.EnactmentList, string, string),
+func (m *Manager) nodeNetworkConfigurationEnactment() (nmstatev1alpha1.NodeNetworkConfigurationEnactment, error) {
+	var enactment nmstatev1alpha1.NodeNetworkConfigurationEnactment
+	pollErr := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		err := m.client.Get(context.TODO(), nmstatev1alpha1.EnactmentKey(m.nodeName, m.policy.Name), &enactment)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				// Let's retry after a while, sometimes it takes some time
```

Comment:
Wait for the enactment to get created should not be considered in the manager. If we need to wait for it explicilty (and Create is not enough), it should happen after we call Create. Adding polling to be "safer" around the codebase makes the application less predictable. We should try to make it as sequential as possible.

Reply:
I always forget that knmstate is distributed we will read the anactments at the same node we create, except for overall policy, let me move it.